include/cetl/pmr/interface_ptr.hpp

```diff
@@ -0,0 +1,92 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef CETL_PMR_INTERFACE_PTR_H_INCLUDED
+#define CETL_PMR_INTERFACE_PTR_H_INCLUDED
+
+#ifndef CETL_H_ERASE
+#    include "cetl/cetl.hpp"
+#endif
+
+#include "cetl/unbounded_variant.hpp"
+
+#include <memory>
+#include <functional>
+
+namespace cetl
+{
+namespace pmr
+{
+
+template <typename Interface>
+class PmrInterfaceDeleter final
```

Comment:
Note - there is no mention of `PmrAllocator` or its nested `value_type` (aka Concrete) types in the template params. This is crucial to make deleter "concrete type erased".